### PR TITLE
Add experimental Python support for automation scripts

### PR DESCRIPTION
**Pull request by Zlatkovsky**
Add experimental Python support for automation scripts (in addition to CFs, which were already supported previously)

This PR also includes some refactoring for both the previous CF support and the runner-to-heartbeat communication.

### CODE REVIEW

File: packages/common/src/constants.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const SCRIPT_URLS = {
   CUSTOM_FUNCTIONS_RUNNER_DEFAULT:
     'https://appsforoffice.microsoft.com/lib/preview/hosted/custom-functions-runtime.js',
   CUSTOM_FUNCTIONS_RUNNER_WITH_JUPYTER_SUPPORT:
-    'https://exceljupyter.azurewebsites.net/agave/dist/external/office-js-custom-0/custom-functions-runtime.debug.js',
+    'https://exceljupyter.azurewebsites.net/agave/dist/external/office-js-custom-1/custom-functions-runtime.debug.js',
```
Comment (nico-bellante) [inline packages/common/src/constants.ts]: checking that this was intentional

File: packages/common/src/utilities/Jupyter/index.ts [hunk 2]
```diff
@@ -76,8 +76,14 @@ interface JupyterWebSocketMessageInputReplyContent {
   value: string;
 }
 
+interface JupyterWebSocketMessageStreamResultContent {
+  name: string;
+  text: string;
+}
+
 enum JupyterWebSocketMessageType {
```
Comment (nico-bellante) [inline packages/common/src/utilities/Jupyter/index.ts]: nit: why the use of enum instead of `type JupyterWebSocketMessageType = "execute_result" | "stream"`?

File: packages/editor/src/utils/custom-functions/generatePythonCFCode.ts [hunk 3]
```diff
@@ -32,8 +32,10 @@ function injectNamespace(script: string, namespace: string) {
   //    (and don't want to do it in the general case in the original regex, because
   //     a regex with a "g" flag becomes a state-ful object that needs to be reset -- see
   //     https://stackoverflow.com/questions/11477415 for more info)
+  const regex = new RegExp(PythonCFSnippetRegex, 'gm');
+
   return script.replace(
-    new RegExp(PythonCFSnippetRegex, 'gm'),
+    regex,
```
Comment (nico-bellante) [inline packages/editor/src/utils/custom-functions/generatePythonCFCode.ts]: why was this changed?

File: packages/common/src/constants.ts (hunk 1 above)
Comment (Zlatkovsky) [reply packages/common/src/constants.ts]: Yep, it was, for the Custom Functions fix that Shaofeng made.

File: packages/common/src/utilities/Jupyter/index.ts (hunk 2 above)
Comment (Zlatkovsky) [reply packages/common/src/utilities/Jupyter/index.ts]: That's a fair point.  The file was actually written by Shaofeng, adopted from another project.  In fact, at the top of the file, you see:

```
/* TODO: for now, copy-pasted this file from an external source -- so disable tslint temporarily */
/* tslint:disable */
```

But, adding @shaofengzhu for his awareness for the future.

File: packages/editor/src/utils/custom-functions/generatePythonCFCode.ts (hunk 3 above)
Comment (Zlatkovsky) [reply packages/editor/src/utils/custom-functions/generatePythonCFCode.ts]: Just to move the regex to a constant, where it can have a comment next to it.  And so that the call itself looks a tad simpler.
